### PR TITLE
Progress Bar progress text implementation in FluentTheme

### LINKED ISSUE #7255: ProgressBar ShowProgressText not implemented in FluentTheme
I made a Window containing a ProgressBar. My application was using the FluentTheme (which is default for both application templates). I set the property 'ShowProgressText' to true. My progress bar was updating correctly, but no text showed up.

I was wondering why it wasn't working. Eventually I found out that the ControlTemplate was different for Default and Fluent theme:

Default theme (src/Avalonia.Themes.Default/ProgressBar.xaml)
```
<ControlTemplate>
  <Grid>
    <Border Background="{TemplateBinding Background}" BorderBrush="{TemplateBinding BorderBrush}" BorderThickness="{TemplateBinding BorderThickness}" CornerRadius="{TemplateBinding CornerRadius}">
      <Panel>
        <Border Name="PART_Indicator" Background="{TemplateBinding Foreground}" IsVisible="{Binding !IsIndeterminate, RelativeSource={RelativeSource TemplatedParent}}"/>
        <Border Name="PART_IndeterminateIndicator" Background="{TemplateBinding Foreground}" IsVisible="{Binding IsIndeterminate, RelativeSource={RelativeSource TemplatedParent}}"/>
      </Panel>
    </Border>
    <LayoutTransformControl HorizontalAlignment="Center" VerticalAlignment="Center" IsVisible="{Binding ShowProgressText, RelativeSource={RelativeSource TemplatedParent}}" Name="PART_LayoutTransformControl">
      <TextBlock Foreground="{DynamicResource ThemeForegroundBrush}" Text="{Binding Value, RelativeSource={RelativeSource TemplatedParent}, StringFormat={}{0:0}%}" />
    </LayoutTransformControl>
  </Grid>
</ControlTemplate>
```

Fluent Theme (src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml)
```
<ControlTemplate>
  <Border x:Name="ProgressBarRoot" ClipToBounds="True" Background="{TemplateBinding Background}" BorderBrush="{TemplateBinding BorderBrush}" BorderThickness="{TemplateBinding BorderThickness}" CornerRadius="{TemplateBinding CornerRadius}">          
      <Panel>
        <Panel x:Name="DeterminateRoot">
          <Border CornerRadius="{TemplateBinding CornerRadius}" x:Name="PART_Indicator" Margin="{TemplateBinding Padding}" Background="{TemplateBinding Foreground}" />
        </Panel>
        <Panel x:Name="IndeterminateRoot">
          <Border x:Name="IndeterminateProgressBarIndicator" CornerRadius="{TemplateBinding CornerRadius}" Margin="{TemplateBinding Padding}" Background="{TemplateBinding Foreground}" />
          <Border x:Name="IndeterminateProgressBarIndicator2" CornerRadius="{TemplateBinding CornerRadius}" Margin="{TemplateBinding Padding}" Background="{TemplateBinding Foreground}" />
        </Panel>
      </Panel>          
  </Border>
</ControlTemplate>
```

Is this a bug or am I misunderstanding something?

### PR DESCRIPTION
**Pull request by Abdesol**
## What does the pull request do?
Add a progress text to a progress bar in fluent theme.

## What is the current behavior?
ProgressBar ShowProgressText doesn't show the progress text.

## What is the updated/expected behavior with this PR?
If the progress par ShowProgressText is set to true, it will show the progress text/status.

## Fixed issues
Fixes https://github.com/AvaloniaUI/Avalonia/issues/7255

### CODE REVIEW

**Comment by Abdesol** [comment]
This images show the previous behaviour before fix and after the fix.
image

**Comment by robloo** [comment]
What happens to the percentage text color when progress is >=50%. Seems like it wouldn't be visible.

1. I think  lightweight resource and potentially another brush property are needed to control this.
2. A new property should be added to disable this without re-templating.

**Comment by maxkatz6** [comment]
> A new property should be added to disable this without re-templating.

Why? ShowProgressText is already there and disabled by default.

File: src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml [hunk 1]
```diff
@@ -21,6 +21,7 @@
     <Setter Property="Template">
       <ControlTemplate>
         <Border x:Name="ProgressBarRoot" ClipToBounds="True" Background="{TemplateBinding Background}" BorderBrush="{TemplateBinding BorderBrush}" BorderThickness="{TemplateBinding BorderThickness}" CornerRadius="{TemplateBinding CornerRadius}">          
+          <Grid>
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml:24]: If there is no row or columns, better to go with Panel instead. Less overhead.

File: src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml [hunk 2]
```diff
@@ -29,7 +30,11 @@
                 <Border x:Name="IndeterminateProgressBarIndicator" CornerRadius="{TemplateBinding CornerRadius}" Margin="{TemplateBinding Padding}" Background="{TemplateBinding Foreground}" />
                 <Border x:Name="IndeterminateProgressBarIndicator2" CornerRadius="{TemplateBinding CornerRadius}" Margin="{TemplateBinding Padding}" Background="{TemplateBinding Foreground}" />
               </Panel>
-            </Panel>          
+            </Panel>
+            <LayoutTransformControl HorizontalAlignment="Center" VerticalAlignment="Center" IsVisible="{Binding ShowProgressText, RelativeSource={RelativeSource TemplatedParent}}" x:Name="PART_LayoutTransformControl">
+              <TextBlock Foreground="{DynamicResource ThemeForegroundBrush}" Text="{Binding Value, RelativeSource={RelativeSource TemplatedParent}, StringFormat={}{0:0}%}" />
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml:35]: ThemeForegroundBrush resource doesn't exist in Fluent theme.

**Comment by robloo** [comment]
> Why? ShowProgressText is already there and disabled by default.

Did not know that, sorry

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0017616-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json)

File: src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml (hunk 1 above)
Comment (Abdesol) [reply src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml:24]: > What happens to the percentage text color when progress is >=50%. Seems like it wouldn't be visible.
> 
> 1. I think  lightweight resource and potentially another brush property are needed to control this.
> 2. A new property should be added to disable this without re-templating.

The Default theme has a similar issue. I will try to make a new pr to fix both of them in the near future.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0017620-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json)

File: src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml (hunk 2 above)
Comment (Abdesol) [reply src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml:35]: > > A new property should be added to disable this without re-templating.
> 
> Why? ShowProgressText is already there and disabled by default.
> 
> @Abdesol default progress bar is somewhat 4px height, right? Do you need to adjust height as well if you set ShowProgressText to true or is it automatically done by current layout?

It is automatically done by the current layout.
See the below images. The height of the progress bar is not set so it is the default one but it is increasing on ShowProgressText = true.

ShowProgressText = false
image


ShowProgressText = true
image

Comment (Abdesol) [reply src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml:35]: > ThemeForegroundBrush resource doesn't exist in Fluent theme.

Ohok... Fixed it now by changing it to `Foreground="{TemplateBinding Foreground}"`

File: src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml (hunk 1 above)
Comment (Abdesol) [reply src/Avalonia.Themes.Fluent/Controls/ProgressBar.xaml]: > If there is no row or columns, better to go with Panel instead. Less overhead.

Fixed that on the current commit.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0017621-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json)

**Review by tsukuyomm — APPROVED** [review]
looking good

**Comment by maxkatz6** [comment]
@Abdesol in default theme text is written with default text foreground (white/black depending on dark mode). Probably it makes sense to do the same here? 
Resource for text color is SystemControlForegroundBaseHighBrush.

**Comment by robloo** [comment]
> @Abdesol in default theme text is written with default text foreground (white/black depending on dark mode). Probably it makes sense to do the same here?
Resource for text color is SystemControlForegroundBaseHighBrush

To fix the issue consistently, text color should be the same as with the ToggleButton which faces the same issue. I would just double check but the SystemControlForegroundBaseHighBrush should be the correct resource.

**Comment by Abdesol** [comment]
> @Abdesol in default theme text is written with default text foreground (white/black depending on dark mode). Probably it makes sense to do the same here? Resource for text color is SystemControlForegroundBaseHighBrush.

oh yeah. Changed that on the current commit.
Sorry for a lot of commits : (
I am not used to the resources

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0017634-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json)

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**